pilot/pkg/networking/core/v1alpha3/cluster_builder_test.go

```diff
@@ -3062,3 +3066,118 @@ func TestApplyDestinationRuleOSCACert(t *testing.T) {
+		connectionPool      *networking.ConnectionPoolSettings
+		expectedHTTPPOpt    *http.HttpProtocolOptions
+	}{
+		{
+			name:    "only update IdleTimeout",
+			cluster: &cluster.Cluster{Name: "foo", ClusterDiscoveryType: &cluster.Cluster_Type{Type: cluster.Cluster_EDS}},
+			httpProtocolOptions: &http.HttpProtocolOptions{
+				CommonHttpProtocolOptions: &core.HttpProtocolOptions{
+					IdleTimeout: gogo.DurationToProtoDuration(&types.Duration{
+						Seconds: 10,
+					}),
+					MaxRequestsPerConnection: &wrappers.UInt32Value{Value: 10},
+				},
+			},
+			connectionPool: &networking.ConnectionPoolSettings{
+				Http: &networking.ConnectionPoolSettings_HTTPSettings{
+					IdleTimeout: &types.Duration{
+						Seconds: 22,
+					},
+				},
+			},
+			expectedHTTPPOpt: &http.HttpProtocolOptions{
+				CommonHttpProtocolOptions: &core.HttpProtocolOptions{
+					IdleTimeout: gogo.DurationToProtoDuration(&types.Duration{
+						Seconds: 22,
+					}),
+					MaxRequestsPerConnection: &wrappers.UInt32Value{Value: 10},
+				},
+			},
+		},
+		{
+			name:    "only update MaxRequestsPerConnection ",
+			cluster: &cluster.Cluster{Name: "foo", ClusterDiscoveryType: &cluster.Cluster_Type{Type: cluster.Cluster_EDS}},
+			httpProtocolOptions: &http.HttpProtocolOptions{
+				CommonHttpProtocolOptions: &core.HttpProtocolOptions{
+					IdleTimeout: gogo.DurationToProtoDuration(&types.Duration{
+						Seconds: 10,
+					}),
+					MaxRequestsPerConnection: &wrappers.UInt32Value{Value: 10},
+				},
+			},
+			connectionPool: &networking.ConnectionPoolSettings{
+				Http: &networking.ConnectionPoolSettings_HTTPSettings{
+					MaxRequestsPerConnection: 22,
+				},
+			},
+			expectedHTTPPOpt: &http.HttpProtocolOptions{
+				CommonHttpProtocolOptions: &core.HttpProtocolOptions{
+					IdleTimeout: gogo.DurationToProtoDuration(&types.Duration{
+						Seconds: 10,
+					}),
+					MaxRequestsPerConnection: &wrappers.UInt32Value{Value: 22},
+				},
+			},
+		},
+		{
+			name:    "update MaxRequestsPerConnection and IdleTimeout",
+			cluster: &cluster.Cluster{Name: "foo", ClusterDiscoveryType: &cluster.Cluster_Type{Type: cluster.Cluster_EDS}},
+			httpProtocolOptions: &http.HttpProtocolOptions{
+				CommonHttpProtocolOptions: &core.HttpProtocolOptions{
+					IdleTimeout: gogo.DurationToProtoDuration(&types.Duration{
+						Seconds: 10,
+					}),
+					MaxRequestsPerConnection: &wrappers.UInt32Value{Value: 10},
+				},
+			},
+			connectionPool: &networking.ConnectionPoolSettings{
+				Http: &networking.ConnectionPoolSettings_HTTPSettings{
+					IdleTimeout: &types.Duration{
+						Seconds: 22,
+					},
+					MaxRequestsPerConnection: 22,
+				},
+			},
+			expectedHTTPPOpt: &http.HttpProtocolOptions{
+				CommonHttpProtocolOptions: &core.HttpProtocolOptions{
+					IdleTimeout: gogo.DurationToProtoDuration(&types.Duration{
+						Seconds: 22,
+					}),
+					MaxRequestsPerConnection: &wrappers.UInt32Value{Value: 22},
+				},
+			},
+		},
+	}
+
+	g := gomega.NewGomegaWithT(t)
```

Comment:
needs to be inside `t.Run` or it will panic if it were to fail. Although we don't need this, can/should use `assert.Equals()`

Reply:
This is a very suggestion, before I didn't find istio has its own `assert` pkg!